src/agent/agent.c

```diff
@@ -511,6 +511,17 @@ bool agent_parse_config(Agent *agent, const char *configfile) {
         value = cfg_get_value(agent->config, CFG_NODE_NAME);
         if (value) {
                 if (!agent_set_name(agent, value)) {
+                         return false;
+                }
+        } else {
+                _cleanup_free_ char *hostname = malloc0(BUFSIZ);
+                if (gethostname(hostname, BUFSIZ) < 0) {
```

Comment:
I think the function to get the default node name should be moved to `protocol.h` next to the definition of [the default port](https://github.com/containers/hirte/blob/main/src/libhirte/common/protocol.h#L6).   

Similar to the default port, it can be called in [agent_new()](https://github.com/containers/hirte/blob/main/src/agent/agent.c#L323) to set the initial value.